src/plugins/data/public/ui/saved_query_form/helpers.tsx

```diff
@@ -0,0 +1,312 @@
+  savedQueryService: SavedQueryService;
+  onSave: (savedQueryMeta: SavedQueryMeta) => void;
+  onClose: () => void;
+  formUiType: 'Modal' | 'Flyout';
+  showFilterOption: boolean | undefined;
+  showTimeFilterOption: boolean | undefined;
+  onSaveAsNew?: () => void;
+}
+
+export interface SavedQueryMeta {
+  title: string;
+  description: string;
+  shouldIncludeFilters: boolean;
+  shouldIncludeTimefilter: boolean;
+}
+
+export function useSaveQueryFormContent({
+  savedQuery,
+  savedQueryService,
+  onSave,
+  onClose,
+  showFilterOption = true,
+  showTimeFilterOption = true,
+  formUiType,
+  onSaveAsNew,
+}: Props) {
+  const [title, setTitle] = useState('');
+  const [enabledSaveButton, setEnabledSaveButton] = useState(false);
+  const [description, setDescription] = useState('');
+  const [savedQueries, setSavedQueries] = useState<SavedQuery[]>([]);
+  const [shouldIncludeFilters, setShouldIncludeFilters] = useState(true);
+  // Defaults to false because saved queries are meant to be as portable as possible and loading
+  // a saved query with a time filter will override whatever the current value of the global timepicker
+  // is. We expect this option to be used rarely and only when the user knows they want this behavior.
+  const [shouldIncludeTimefilter, setIncludeTimefilter] = useState(false);
+  const [formErrors, setFormErrors] = useState<string[]>([]);
+
+  // Instead of setting the default values based on the saved query above,
+  // we use this effect so that in case when user select "Save as new" in the flyout,
+  // the initial state resets with change to 'savedQuery'
+  useEffect(() => {
+    setTitle(savedQuery ? savedQuery.title : '');
+    setEnabledSaveButton(Boolean(savedQuery));
+    setDescription(savedQuery ? savedQuery.description : '');
+    setShouldIncludeFilters(savedQuery ? !!savedQuery.filters : true);
+    setIncludeTimefilter(savedQuery ? !!savedQuery.timefilter : false);
+    setFormErrors([]);
+  }, [savedQuery]);
+
+  const titleConflictErrorText = i18n.translate(
+    'data.search.searchBar.savedQueryForm.titleConflictText',
+    {
+      defaultMessage: 'Name conflicts with an existing saved query',
+    }
+  );
+
+  const savedQueryDescriptionText = i18n.translate(
+    'data.search.searchBar.savedQueryDescriptionText',
+    {
+      defaultMessage: 'Save query text and filters that you want to use again.',
+    }
+  );
+
+  useEffect(() => {
+    const fetchQueries = async () => {
+      const allSavedQueries = await savedQueryService.getAllSavedQueries();
+      const sortedAllSavedQueries = sortBy(allSavedQueries, 'attributes.title') as SavedQuery[];
+      setSavedQueries(sortedAllSavedQueries);
+    };
+    fetchQueries();
+  }, [savedQueryService]);
+
+  const validate = useCallback(() => {
+    const errors = [];
+    if (
+      !!savedQueries.find(
+        (existingSavedQuery) => !savedQuery && existingSavedQuery.attributes.title === title
+      )
+    ) {
```

Comment:
```suggestion
    if (
      !savedQuery && 
      savedQueries.some(
        (existingSavedQuery) => existingSavedQuery.attributes?.title === title
      )
    ) {
```

Reply:
Are we checking title elsewhere?

Reply:
Yes, if title is undefined the save button stays disabled